dfget/util/algorithm_test.go

```diff
@@ -52,4 +52,10 @@ func (s *AlgorithmTestSuite) TestShuffle(c *check.C) {
 		Shuffle(n, func(i, j int) { isRun++ })
 		c.Assert(isRun, check.Equals, n-1)
 	}
+
+	Shuffle(1<<31, func(i, j int) {
+		if 31-1 <= i {
```

Comment:
Could you please add some description here? I'm not very clear about its meaning of this.

Reply:
Because if i let the loop complete, it will time out.
So i need to make the loop stop early.

Reply:
sorry, this is my mistake. I should be 1<<31-1.